src/worker/mod.rs

```diff
@@ -141,7 +141,21 @@ impl Worker {
             config,
         }
     }
-    pub(crate) async fn shutdown(self) {
+
+    /// Tell the worker to begin the shutdown process. Can be used before [await_shutdown] if
+    /// polling should be ceased before it is possible to consume the worker instance.
+    pub(crate) fn notify_shutdown(&self) {
+        self.wf_task_poll_buffer.notify_shutdown();
+        if let Some(sq) = self.sticky_queue.as_ref() {
+            sq.poll_buffer.notify_shutdown();
+        }
+        if let Some(b) = self.at_task_poll_buffer.as_ref() {
+            b.notify_shutdown();
+        }
+    }
+
+    /// Resolves when shutdown of the worker is complete
+    pub(crate) async fn await_shutdown(self) {
```

Comment:
Maybe rename this to `shutdown_complete` since it doesn't actually await?
```rust
worker.shutdown_complete().await
```
as opposed to
```rust
worker.await_shutdown().await
```